esignet-core/src/main/java/io/mosip/esignet/core/spi/PublicKeyRegistryService.java

```diff
@@ -7,10 +7,11 @@
 
 import io.mosip.esignet.core.dto.PublicKeyRegistry;
 
+import java.time.LocalDateTime;
 import java.util.Optional;
 
 public interface PublicKeyRegistryService {
 
     Optional<PublicKeyRegistry> findLatestPublicKeyByPsuTokenAndAuthFactor(String psuToken, String authFactor);
-
+    Optional<PublicKeyRegistry> findFirstByIdHashAndThumbprintAndExpiredtimesGreaterThanOrderByExpiredtimesDesc(String idHash, String thumbPrint);
```

Comment:
you can rename the method in service to a logical name, it doesn't need to match the repository method name